shell/platform/linux/fl_engine.cc

```diff
@@ -575,7 +636,18 @@ gboolean fl_engine_start(FlEngine* self, GError** error) {
   if (self->embedder_api.RunsAOTCompiledDartCode()) {
     FlutterEngineAOTDataSource source = {};
     source.type = kFlutterEngineAOTDataSourceTypeElfPath;
-    source.elf_path = fl_dart_project_get_aot_library_path(self->project);
+    std::string patch_path;
+    auto setup_shorebird_result =
+        fl_set_up_shorebird(args.assets_path, patch_path);
+    if (setup_shorebird_result) {
+      // If we have a patch installed, we replace the default AOT library path
+      // with the patch path here.
+      source.elf_path = patch_path.c_str();
```

Comment:
Does this take ownership of the c_str()?  (presumably by copying it into a std::string?). If elf_path is just a char* this will crash sometimes.